transforms/helpers/validation-helper.ts

```diff
@@ -61,20 +63,24 @@ export function hasValidProps(
 
   let errors: string[] = [];
   for (const instanceProp of instanceProps) {
-    if (!classFields && instanceProp.type === 'Literal') {
-      errors.push(`[${instanceProp.name}]: Need option '--class-fields=true'`);
-    }
+    if (instanceProp instanceof EOSimpleProp) {
+      if (!classFields) {
+        errors.push(
+          `[${instanceProp.name}]: Need option '--class-fields=true'`
+        );
+      }
 
-    if (
-      instanceProp.type === 'ObjectExpression' &&
-      !['actions', 'queryParams'].includes(instanceProp.name)
-    ) {
-      errors.push(
-        `[${instanceProp.name}]: Transform not supported - value is of type object. For more details: eslint-plugin-ember/avoid-leaking-state-in-ember-objects`
-      );
+      if (
+        instanceProp.type === 'ObjectExpression' &&
+        instanceProp.name !== 'queryParams'
```

Comment:
I still find it *very* weird that this is hardcoded. Instead of hardcoding it, what do you think
about making it an option to `decorator` (see my suggestion in `#2`) and defaulting it to
`['queryParams']`.

```json5
{
  decorators: {
    ignorePropertiesNamed: ['queryParams']
  }
}
```

If you do this, make sure to document that if you explicitly override this option, you need to
include `queryParams` if you want the default behavior.

(I was going to say automatically include it, but that opens up a can of worms about how to remove
it if you really don't want it.)